CMakeLists.txt

```
@@ -167,3 +167,25 @@ function(mbed_generate_bin_hex target)
         VERBATIM
     )
 endfunction()
+
+#
+# Parse toolchain generated map file for this `target` and display readable table format
+#
+function(mbed_parse_map_file target)
+    if(MBED_GENERATE_MAP_TABLE)
+        find_package (Python3)
+        add_custom_command(
+            TARGET ${target} POST_BUILD
+            COMMAND ${Python3_EXECUTABLE} ${MBED_ROOT}/tools/memap.py -t ${MBED_TOOLCHAIN} -d 4 ${CMAKE_BINARY_DIR}/${target}${CMAKE_EXECUTABLE_SUFFIX}.map
+            WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+            COMMENT "Displaying memory map for ${target}")
```

The indentation style used to improve readability is:
```suggestion
        add_custom_command(
            TARGET
                ${target}
            POST_BUILD
                COMMAND ${Python3_EXECUTABLE} ${MBED_ROOT}/tools/memap.py -t ${MBED_TOOLCHAIN} -d 4 ${CMAKE_BINARY_DIR}/${target}${CMAKE_EXECUTABLE_SUFFIX}.map
            WORKING_DIRECTORY
                ${CMAKE_BINARY_DIR}
            COMMENT
                "Displaying memory map for ${target}"
)
```